### DISCUSSION TITLE
Status StartAsync stopping program, not working

### DISCUSSION THREAD

**Discussion by ghost** [spectreconsole/spectre.console#1420] [General]
**Information**
 - OS: Linux
 - Version: 0.48.0
 - Terminal: VSCode Integrated terminal / Alacritty

**Describe the bug**
While trying to run AnsiConsole.Status().StartAsync(), the spinner doesn't show up, and after a CliWrap function finishes executing, the program exits with no exceptions or prints, despite there being more prints inside the Status function, as well as outside of it.

**To Reproduce**
Add latest CliWrap and Spectre.Console packages to a .NET 8 project. In a separate function that is called from Main(), put
```cs
        var output = await AnsiConsole.Status().StartAsync("Generating files", async ctx =>
        {  
            ctx.Spinner(Spinner.Known.Ascii);
            AnsiConsole.WriteLine("Hey, Wilbrand! Can you help me out real quick?");
            BufferedCommandResult? result = null;
            try {
                result = await Cli.Wrap(Path.Join(Directory.GetParent(Environment.ProcessPath).FullName, "/extra/wilbrand"))
            .WithArguments(new[] { macaddr, DateTime.Now.ToShortDateString(), version, sd })
            .ExecuteBufferedAsync(); //program seems to exit here? no more prints after this
            } catch (Exception e) {
                AnsiConsole.WriteLine(e.ToString());
            }
            
            await Task.Delay(5000);
            if (result.StandardError == "") {
                AnsiConsole.WriteLine("Thanks bro!");
            } else {
                AnsiConsole.MarkupLine("[red]It's alright if you messed up, you tried your best.[/]\n\n[red dim]Try again, maybe?[/]");
                Environment.Exit(1);
            }
            return result.StandardOutput;
        });
        AnsiConsole.Markup($"[grey dim italics]{output}");
        AnsiConsole.MarkupLine("[bright green]Done! Check your SD card for a surprise ;)");
```
And then run the program, with `macaddr`, `version`, and `sd` variables being strings.
**Expected behavior**
A spinner to appear as the external program is run, then the output of the program to be printed in the console, as well as the three prints that come after the CliWrap function.

**Screenshots**
![image](https://github.com/spectreconsole/spectre.console/assets/83098445/eda83074-1934-4a02-84fd-84e70afd426b)

**Additional context**
N/A

**Comment by nils-a** [comment]
My guess is, it has something to do with that "wildbrand" tool. 
If I replace the call to some other executable that is available on my machine, the code runs fine. 

(apart from the `italics` markup that should be named `italic`, `bright` does not exist, and you should call `EscapeMarkup()` on the `output` when writing it to the console.)

My output:

```text
Hey, Wilbrand! Can you help me out real quick?
Thanks bro!
[... output of the executable ...]
Done! Check your SD card for a surprise ;)
```

**Reply by ghost** [reply]
Right, but it doesn't seem to just be due to 'wildbrand', as I tried just removing the call to Cli.Wrap and nothing showed up still. Even if it was due to the executable, what would cause it? The tool sends some messages to stdout, generates some files, then exits normally.

**Reply by ghost** [reply]
I think I just solved the problem, I was incorrectly using async/await without returning Tasks, and I didn't make the Main method async, which I assume made a ton of issues. It works great now though, thanks! I'm not that advanced with async/await, haha
